tools/optimise_localisation.py

```diff
@@ -0,0 +1,246 @@
+        return total_rmse
+
+    except subprocess.TimeoutExpired:
+        print("Benchmark timed out.")
+        return None
+
+    except subprocess.CalledProcessError as e:
+        print("An error occurred while running the benchmark:")
+        print(e.stderr)
+        return None
+
+
+def objective(trial):
+    """
+    The objective function for Optuna that defines the optimization problem.
+    """
+    field_line_distance_weight = trial.suggest_float("field_line_distance_weight", 1e-2, 1e2, log=True)
+    field_line_intersection_weight = trial.suggest_float("field_line_intersection_weight", 1e-2, 1e2, log=True)
+    state_change_weight = trial.suggest_float("state_change_weight", 1e-2, 1e2, log=True)
+    goal_post_distance_weight = trial.suggest_float("goal_post_distance_weight", 1e-2, 1e2, log=True)
+    change_limit_x = trial.suggest_float("change_limit_x", 1e-2, 1.0)
+    change_limit_y = trial.suggest_float("change_limit_y", 1e-2, 1.0)
+    change_limit_theta = trial.suggest_float("change_limit_theta", 1e-2, 1.0)
+    max_association_distance = trial.suggest_float("max_association_distance", 1e-2, 10.0)
+    Q_x = trial.suggest_float("Q_x", 1e-6, 1e-2, log=True)
+    Q_y = trial.suggest_float("Q_y", 1e-6, 1e-2, log=True)
+    Q_theta = trial.suggest_float("Q_theta", 1e-6, 1e-2, log=True)
+    R_x = trial.suggest_float("R_x", 1e-2, 1e1)
+    R_y = trial.suggest_float("R_y", 1e-2, 1e1)
+    R_theta = trial.suggest_float("R_theta", 1e-2, 1e1)
+
+    params = {
+        "field_line_distance_weight": field_line_distance_weight,
+        "field_line_intersection_weight": field_line_intersection_weight,
+        "state_change_weight": state_change_weight,
+        "goal_post_distance_weight": goal_post_distance_weight,
+        "change_limit_x": change_limit_x,
+        "change_limit_y": change_limit_y,
+        "change_limit_theta": change_limit_theta,
+        "max_association_distance": max_association_distance,
+        "Q_x": Q_x,
+        "Q_y": Q_y,
+        "Q_theta": Q_theta,
+        "R_x": R_x,
+        "R_y": R_y,
+        "R_theta": R_theta,
+    }
+
+    modify_yaml(params, LOCALISATION_YAML_PATH)
+
+    # Run the benchmark with user attribute nbs_file
+    rmse = run_benchmark(trial.study.user_attrs["nbs_file"])
+
+    if rmse is None:
+        return float("inf")
+
+    trial.set_user_attr("rmse_translation", parse_rmse_translation)
+    trial.set_user_attr("rmse_rotation", parse_rmse_rotation)
+
+    return rmse
+
+
+@run_on_docker
+def register(parser):
+    """
+    Register command-line arguments for the optimization tool.
+    """
+    parser.description = "Optimises YAML configuration parameters for the LocalisationBenchmark using Optuna."
+    parser.add_argument("--n_trials", type=int, default=1000, help="Number of optimization trials to run.")
+    parser.add_argument("--nbs_file", type=str, default=NBS_FILE, help="Path to the .nbs file with ground truth data.")
```

Comment:
I feel like this should be a required argument - I'd expect most people wouldn't be renaming the generated nbs file